Uplift/Views/Headers/CircularProgressView.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  CircularProgressView.swift
+//  Uplift
+//
+//  Created by elvis on 10/18/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+class CircularProgressView: UIView {
+
+    private var progressLayer = CAShapeLayer()
+    private var trackLayer = CAShapeLayer()
+
+    private var progressColor: UIColor!
+    private var width: CGFloat!
+
+    init(progressColor: UIColor, width: CGFloat) {
+        super.init(frame: .zero)
+
+        self.progressColor = progressColor
+        self.width = width
+
+        createCircularPath()
+    }
+
+    private func createCircularPath() {
+        self.layer.cornerRadius = width!/2
+        let circularPath = UIBezierPath(arcCenter: CGPoint(x: width/2, y: width/2) , radius: (width - 1.5)/2, startAngle: CGFloat(-0.5 * .pi), endAngle: CGFloat(2.0 * .pi), clockwise: true)
+
```

Comment:
Is this going to be used for the capacities? If so, I think it would be cool to make this more generic maybe with a `percent`, `start angle`, and`linewidth` (and maybe animated??). Ofc that's something that could be done later as well.